src/main/java/org/alliancegenome/curation_api/services/base/BaseDTOCrudService.java

```diff
@@ -9,7 +9,8 @@
 
 	protected abstract void init();
 
-	public abstract E upsert(T dto) throws ObjectUpdateException;
+	public abstract E upsert(T dto, String dataType) throws ObjectUpdateException;
+
```

Comment:
@markquintontulloch so what would be better is to create two methods one with the params and one without. That way you don't have to change as much code, and don't have to pass nulls all over the place. As in:
```java
public abstract E upsert(T dto) throws ObjectUpdateException {
     return upsert(dto, null);
}
```

Reply:
Wouldn't that require implementing both methods in all service classes that extend this?  This will be reverted (along with all the null passing) as soon as we implement the LinkML changes in the Java code and DQMs are supplying the data provider.

Reply:
Thats why I took off the abstract... I realize all this code is going to get ripped out thats why I am suggesting changes that change the least amount of code.